habitat_baselines/common/base_trainer.py

```diff
@@ -150,13 +150,58 @@ class BaseRLTrainer(BaseTrainer):
     device: torch.device  # type: ignore
     config: Config
     video_option: List[str]
+    num_updates_done: int
+    num_steps_done: int
     _flush_secs: int
+    _last_checkpoint_percent: float
 
     def __init__(self, config: Config) -> None:
         super().__init__()
         assert config is not None, "needs config file to initialize trainer"
         self.config = config
         self._flush_secs = 30
+        self.num_updates_done = 0
+        self.num_steps_done = 0
+        self._last_checkpoint_percent = -1.0
+
+        if config.NUM_UPDATES != -1 and config.TOTAL_NUM_STEPS != -1:
+            raise RuntimeError(
+                "NUM_UPDATES and TOTAL_NUM_STEPS are both specified.  One must be -1.\n"
+                " NUM_UPDATES: {} TOTAL_NUM_STEPS: {}".format(
+                    config.NUM_UPDATES, config.TOTAL_NUM_STEPS
+                )
+            )
+
+        if config.NUM_UPDATES == -1 and config.TOTAL_NUM_STEPS == -1:
+            raise RuntimeError(
+                "One of NUM_UPDATES and TOTAL_NUM_STEPS must be specified.\n"
+                " NUM_UPDATES: {} TOTAL_NUM_STEPS: {}".format(
+                    config.NUM_UPDATES, config.TOTAL_NUM_STEPS
+                )
+            )
+
+    def percent_done(self) -> float:
+        if self.config.NUM_UPDATES != -1:
+            return self.num_updates_done / self.config.NUM_UPDATES
+        else:
+            return self.num_steps_done / self.config.TOTAL_NUM_STEPS
+
+    def is_done(self) -> bool:
+        return self.percent_done() >= 1.0
+
+    def should_checkpoint(self) -> bool:
+        checkpoint_every = 1 / self.config.NUM_CHECKPOINTS
```

Comment:
Can you elaborate more why you decided to use `NUM_CHECKPOINTS` instead of checkpoint every NUM updates? Thank you!
One use case, when it will require some math to be done is when user want to run "endless" training and stop it when the training converged. Then very large number used for NUM_UPDATES/TOTAL_NUM_STEPS. Those will required to calculate NUM_CHECKPOINTS.

Reply:
I found it very annoying to have to calculate how many updates per checkpoint I wanted since the number of frames per update is a factor of number of DD-PPO workers, number of simulators, and rollout length.  I can keep both functionality if you'd like.  I just deleted the other one since I haven't found myself wanting it since I made this change in my fork 8+ months ago.